### ISSUE TITLE
[Bug] DataZoom's miniature misrepresents data in a managed plot (the plot drawn over the slider doesn't match the main plot's data)

### ISSUE THREAD

**Issue by MrCatEsq** [apache/echarts#17122]
### Version

5.3.x

### Link to Minimal Reproduction

https://codepen.io/sdfsdfsdasd/pen/ExQoyzG

### Steps to Reproduce

1. Create a simple [line plot/chart](https://echarts.apache.org/en/option.html#series-line.type) using Apache Echarts 
2.  Add built-in data scaling mechanism: [dataZoom](https://echarts.apache.org/en/option.html#dataZoom).
3. Config and initialization script may be pasted to Echarts [online editor](https://echarts.apache.org/examples/en/editor.html?c=area-simple). Or you can use CodePen link presented above.

```
let x = [];
let y = [];
let scaled = [];
/*y = [
  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 300, 300, 
  10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 0, 0, 0, 0, 0, 0
];*/
for (let i = 1; i < 300; i++) {
  x.push(i);
  element = Math.random() * 40 + 50;
  y.push(element);
  scaled.push(element * 6 - 250);
}
option = {
  xAxis: {
    data: x
  },
  yAxis: {
    min: 0,
    max: 300
  },
  dataZoom: [
    {
      start: 50,
      end: 58.5
    }
  ],
  series: [
    {
      name: 'Fake Data',
      type: 'line',
      symbol: 'none',
      data: y
    },
    {
      name: 'Simulated Scaling',
      type: 'line',
      symbol: 'none',
      lineStyle: {
        opacity: 0.3
      },
      data: scaled
    }
  ]
};
```

### Current Behavior

DataZoom component reaches its main goal, but there is a question to scaled data representation, made by dataZoom. By default, dataZoom doesn't take into account the chart scale [limits](https://echarts.apache.org/en/option.html#yAxis.min) ticks or/and the minimum and maximum allowable values ([range of a function](https://en.wikipedia.org/wiki/Range_of_a_function), represented by the plot). Instead, **the thumbnail of the chart is drawn on the specific value range passed to the plot in [series](https://echarts.apache.org/en/option.html#series) section.** In addition, everytime **a small indent is added from the minimum and maximum values ​​to the borders of the graphic element**.

### Expected Behavior

As a result, the representation of the visualised data looks inconsistent with reality: null is not null, max is not max (because they don't match the lower and higher bounds of the coordinate area of ​​the thumbnail plot, respectively), the amplitude of the chart fluctuations does not correspond to the scale of real data fluctuations:

![datazoom](https://user-images.githubusercontent.com/95284657/159572779-2b5c7772-0071-4477-9f84-bbaca278ec82.png)

As you can see, the magnitude of the fluctuations of the graph, drawn by dataZoom doesn't correspond rather to the main data, but to some kind of artificial transformation of them (light green graph). Then try to comment line 14 and uncomment lines from 6 to 9 (on CodePen). _At the start_ of the plot you'll see main (blue) graph touching y zero line, but not on the thumbnail.

### Environment

```markdown
- OS: Windows 8.1, Windows 10, MacOS X
- Browser: Chrome, Safari, Opera
```

### Any additional comments?

I didn't find any [params](https://echarts.apache.org/en/option.html#dataZoom) for dataZoom that make them to look like expected. My question on StackOverFlow was left [unanswered](https://stackoverflow.com/questions/71574041/apache-echarts-datazooms-miniature-misrepresents-data-in-a-managed-plot-the-p), a [feature request](https://github.com/apache/echarts/issues/16724) posted earlier stays unconfirmed so I made this bug report.

Is there a way (documented or undocumented) to remove the indents and force the plot to use the minimum and maximum values ​​allowed for the yAxis ticks? Some options like `dataZoom.slider.removeGaps: true | false` and `dataZoom.slider.range` (values like "asInData" (current behavior), "asInYaxis" (my case), etc)?

**Comment by tyn1998** [comment]
Hello, @MrCatEsq ~ If you mean this "indent"?

<img width="699" alt="image" src="https://user-images.githubusercontent.com/32434520/170622588-cbc01787-ad9a-4942-b152-17c7e0667484.png">

**Comment by tyn1998** [comment]
hi, @MrCatEsq. I think this is what you might want:

<img width="1191" alt="image" src="https://user-images.githubusercontent.com/32434520/170631942-4996c0a4-cec7-4892-8315-46b1a96e9651.png">

For me, I didn't find any option that will changes this behavior, neither. But I located source code that is related to this:

https://github.com/apache/echarts/blob/d805ff25a2065f21d9a879998cae7075fdfa91e3/src/component/dataZoom/SliderZoomView.ts#L378-L383

And I tried to modify as below and it works:

<img width="1117" alt="image" src="https://user-images.githubusercontent.com/32434520/170632499-12138175-8341-49ed-bf45-3effa301dedd.png">

In fact, maybe @pissang knew that the feature is a wanted one so he left a "FIXME" in code:

https://github.com/apache/echarts/blob/d805ff25a2065f21d9a879998cae7075fdfa91e3/src/component/dataZoom/SliderZoomView.ts#L401-L402

I never touched the source code of Echarts before, so I might be wrong. We'd better wait for core developers to give an anwser.

**Comment by MrCatEsq** [comment]
> Hello, @MrCatEsq ~ If you mean this "indent"?

Hi, @tyn1998! Thanks for the answer. Yes, I mean vertical indents/paddings from top and bottom of the DataZoom border.
Your hotfix _partially solves_ the problem, [this code sample](https://codepen.io/sdfsdfsdasd/pen/eYVVZEm) prooves that. Now the vertical indents have disappeared, and the amplitude of the graph fluctuations corresponds to its representation on the datazoum. 

But we still have problems if [the graph partially has no data](https://codepen.io/sdfsdfsdasd/pen/yLvveQQ).

The fix also not working if I [don't prefilled "y" array](https://codepen.io/sdfsdfsdasd/pen/QWQQNmR).

**Comment by tyn1998** [comment]
Hi, @MrCatEsq, glad to hear it helps a bit. Since the problem is beyond my capability, further help might be sought from the core team :)

**Comment by niko-m** [comment]
Same stuff. The main chart has a scale from 0 to 250 and values from approximately 52.20 to 52.37. The chart thumbnail does not match the main one. It would be great to be able to set the minimum and maximum for the chart thumbnail.

![photo_2022-06-22 11 00 15](https://user-images.githubusercontent.com/13641921/174976808-4ba15e29-f7e7-40bf-acd6-2ea6d9adef3a.jpeg)

> further help might be sought from the core team :)



@pissang, @100pah, @Ovilia, @plainheart, @susiwen8 
May be, someone will find it interesting to make a hotfix in one of the next minor releases?

**Comment by Ovilia** [comment]
This might be fixed in #17143 . Please wait for this PR to be merged.
